src/operator-run.c

```diff
@@ -606,6 +606,11 @@ static void compute_u8softargmax(
 
 enum qnnp_status qnnp_run_operator(qnnp_operator_t op, pthreadpool_t threadpool)
 {
+  // For any ukernel type, there is no work to do if the batch size is 0.
+  if (op->batch_size == 0) {
+    return qnnp_status_success;
```

Comment:
In what case will we receive batch size 0? Should we be emitting an info message here saying we got batch size 0, or is that done in the C2 land?

Reply:
I don't think we should be emitting a message in any case, just like Python doesn't emit a message if you try to do a comprehension over an empty list.